lightningd/invoice.c

```diff
@@ -1207,7 +1209,18 @@ static struct command_result *json_invoice(struct command *cmd,
 	info->b11->min_final_cltv_expiry = *cltv;
 	info->b11->expiry = *expiry;
 	info->b11->description = tal_steal(info->b11, desc_val);
-	info->b11->description_hash = NULL;
+	/* BOLT #11:
+	 * * `h` (23): `data_length` 52. 256-bit description of purpose of payment (SHA256).
+	 *...
+	 * A writer:
+	 *...
+	 *    - MUST include either exactly one `d` or exactly one `h` field.
+	 */
+	if (*hashonly) {
+		info->b11->description_hash = tal(info->b11, struct sha256);
+		sha256(info->b11->description_hash, desc_val, strlen(desc_val));
+	} else
+		info->b11->description_hash = NULL;
```

Comment:
Why not just 
```suggestion
         info->b11->description_hash = NULL;
	/* BOLT #11:
	 * * `h` (23): `data_length` 52. 256-bit description of purpose of payment (SHA256).
	 *...
	 * A writer:
	 *...
	 *    - MUST include either exactly one `d` or exactly one `h` field.
	 */
	if (*hashonly) {
		info->b11->description_hash = tal(info->b11, struct sha256);
		sha256(info->b11->description_hash, desc_val, strlen(desc_val));
	}
		
```

There is any convention or particular code style? Just curius

Reply:
Yes.  Generally I prefer to avoid the pattern of

```
x = foo
if (cond)
    x = bar
```

Because if you forget something, the compiler won't warn you.  For example:

```
if (cond)
    x = bar
else
   y = foo //  Oops, was meant to be x!

print(x) // Compiler warns here that x may not be initialized!
```

Reply:
Cool, thanks!